internal/client/client.go

```diff
@@ -682,6 +682,34 @@ func SyncTimelineHasEventID(roomID string, eventID string) SyncCheckOpt {
 	})
 }
 
+// Check that the state for `roomID` has an event which passes the check function.
+func SyncStateHas(roomID string, check func(gjson.Result) bool) SyncCheckOpt {
```

Comment:
We need to be careful here, because this can be misused. The `state` section of the room contains the state at the start of the timeline, so it's entirely possible that the state to check is in the `timeline` and not the `state`. We should be very careful about checking one or the other. What is the use case here?

Reply:
Ultimately I would like to check that a `lazy_load_members` /sync response includes the membership of a given sender in the `state` section. The sender joined the room before the syncing user and their membership is not expected to appear in the timeline. The planned usage looks like this:
```go
syncRes, _ := alice.MustSync(t,
	client.SyncReq{
		Since:  previousSyncToken,
		Filter: buildLazyLoadingSyncFilter(nil),
	},
)

err = client.SyncTimelineHasEventID(serverRoom.RoomID, event.EventID())(alice.UserID, syncRes)
if err != nil {
	t.Errorf("Did not find message in lazy-loading /sync response: %s", err)
}

err := client.SyncStateHasStateKey(serverRoom.RoomID, "m.room.member", event.Sender())(alice.UserID, syncRes)
if err != nil {
	t.Errorf("Did not find %s's m.room.member event in lazy-loading /sync response: %s", event.Sender(), err)
}
```

I'm avoiding `MustSyncUntil` because it wouldn't fail immediately if `event` appears in the timeline without the corresponding sender membership, and the message is expected to already be available via /sync.